lib/lotus/commands/generate/action.rb

```diff
@@ -0,0 +1,237 @@
+require 'lotus/commands/generate/abstract'
+require 'lotus/generators/generator'
+require 'lotus/utils/string'
+require 'lotus/routing/route'
+
+module Lotus
+  module Commands
+    class Generate < Thor
+      class Action < Abstract
+
+        # @since x.x.x
+        # @api private
+        ACTION_SEPARATOR = /\/|\#/
+
+        # @since x.x.x
+        # @api private
+        ROUTE_ENDPOINT_SEPARATOR = '#'.freeze
+
+        # @since x.x.x
+        # @api private
+        QUOTED_NAME = /(\"|\'|\\)/
+
+        # Default HTTP method used when generating an action.
+        # @since x.x.x
+        # @api private
+        DEFAULT_HTTP_METHOD = 'GET'.freeze
+
+
+        def initialize(options, application_name, controller_and_action_name)
+          super(options)
+          if !environment.container?
+            application_name = File.basename(Dir.pwd)
+          end
+
+          controller_and_action_name = Utils::String.new(controller_and_action_name).underscore.gsub(QUOTED_NAME, '')
+
+          controller_name, action_name = controller_and_action_name.split(ACTION_SEPARATOR)
+
+          @application_name = Utils::String.new(application_name).classify
+          @controller_name = Utils::String.new(controller_name).classify
+          @action_name = Utils::String.new(action_name).classify
+
+          @generator = Lotus::Generators::Generator.new(template_source_path, base_path)
+
+          assert_application_name!
+          assert_controller_name!
+          assert_action_name!
+          assert_http_method!
+          assert_test_type!
+        end
+
```

Comment:
i'd be interested in feedback on my "do we need the app architecture at all". This could simplify the action generator. There is also a method  ```def thing_that_needs_a_name``` that is in need of a better name. Which could again be avoided if the "app" architecture is not needed.